lib/spack/env/cc

```diff
@@ -98,25 +98,25 @@ case "$command" in
     cpp)
         mode=cpp
         ;;
-    cc|c89|c99|gcc|clang|icc|pgcc|xlc)
+    cc|c89|c99|gcc|clang|icc|pgcc|xlc|xlc_r)
```

Comment:
I have the following changes that list the fflags in the flags section:
diff --git a/lib/spack/spack/compilers/__init__.py b/lib/spack/spack/compilers/__init__.py
index 6e65f50..7e530f0 100644
--- a/lib/spack/spack/compilers/__init__.py
+++ b/lib/spack/spack/compilers/__init__.py
@@ -40,6 +40,7 @@ from spack.util.naming import mod_to_class
 
 _imported_compilers_module = 'spack.compilers'
 _path_instance_vars = ['cc', 'cxx', 'f77', 'fc']
+_flags_instance_vars = ['cflags', 'cxxflags', 'fflags']
 _other_instance_vars = ['modules', 'operating_system', 'environment',
                         'extra_rpaths']
 _cache_config_file = []
@@ -60,6 +61,8 @@ def _to_dict(compiler):
     d['paths'] = dict((attr, getattr(compiler, attr, None))
                       for attr in _path_instance_vars)
     d['flags'] = dict((fname, fvals) for fname, fvals in compiler.flags)
+    d['flags'].update(dict((attr, getattr(compiler, attr, None))
+                      for attr in _flags_instance_vars if hasattr(compiler, attr)))
     d['operating_system'] = str(compiler.operating_system)
     d['target'] = str(compiler.target)
     d['modules'] = compiler.modules if compiler.modules else []

With these small change, the compilers.yaml file looks as follows:

- compiler:
    environment: {}
    extra_rpaths: []
    flags:
      fflags: -qzerosize -qfixed
    modules: []
    operating_system: rhel7
    paths:
      cc: /usr/bin/xlc
      cxx: /usr/bin/xlc++
      f77: /usr/bin/xlf
      fc: /usr/bin/xlf2003
    spec: xl@13.1
    target: ppc64le
- compiler:
    environment: {}
    extra_rpaths: []
    flags:
      fflags: -qzerosize -qfixed
    modules: []
    operating_system: rhel7
    paths:
      cc: /usr/bin/xlc_r
      cxx: /usr/bin/xlc++_r
      f77: /usr/bin/xlf_r
      fc: /usr/bin/xlf2003_r
    spec: xl_r@13.1
    target: ppc64le

Withe the fflags showing under flags.  What is not clear to me is what are the other flags that should go under there?  The compiler classes define other flags, like openmp_flag, pic_flag, cxx11_flag.  They are not listed at all.